### PR TITLE
React hook form in the questionnaire

### PR DESCRIPTION
**Pull request by pkrucz00**
This PR adds `react-hook-form` to the after-call questionnaire.

The code in the target branch has only the needed appearance. This adds a little bit of functionality to the questionnaire by:
  -  defining the type that the questionnaire should return
  - adding `react-hook-form` to deps
  - wrapping the components so they are compliant with `react-hook-form`
  - adding the components to the questionnaire
  - validations

### CODE REVIEW

File: assets/src/features/leaving-page/components/Questionnaire.tsx [hunk 1]
```diff
@@ -2,17 +2,29 @@ import React, { FC, useState } from "react";
...
 
 type CommentBoxProps = {
   isOpen: boolean;
   setOpen: () => void;
+  register: UseFormRegister<Inputs>;
+  error: FieldError | undefined;
 };
 
-const CommentBox = ({ isOpen, setOpen }: CommentBoxProps) => {
+const CommentBox = ({ isOpen, setOpen, register, error }: CommentBoxProps) => {
```
Comment (pkrucz00) [inline assets/src/features/leaving-page/components/Questionnaire.tsx:27]: To different file

File: assets/src/features/leaving-page/components/Questionnaire.tsx [hunk 2]
```diff
@@ -33,45 +52,100 @@ type QuestionnaireProps = {
...
+  } = useForm<Inputs>();
+  const emailInput = watch("email");
+  const emailFilled = !!emailInput && !errors.email;
+
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
+    console.log(data);
+    onSubmitClick();
+  };
+
+  const ratingNames: RatingName[] = ["video", "audio", "screenshare"];
```
Comment (pkrucz00) [inline assets/src/features/leaving-page/components/Questionnaire.tsx:71]: Can be inlined

File: assets/src/features/leaving-page/components/Questionnaire.tsx [hunk 3]
```diff
@@ -33,45 +52,100 @@ type QuestionnaireProps = {
...
+                required
+              />
+            )}
+          />
         </div>
         <div aria-label="questionnaire-submit" className="flex flex-col content-center gap-4">
           <Button
-            onClick={onSubmitClick}
+            type="submit"
+            onClick={() => {}} //noop here because otherwise it directs to the homepage
```
Comment (pkrucz00) [inline assets/src/features/leaving-page/components/Questionnaire.tsx:136]: Change the default `onClick` to noop

File: assets/src/features/shared/components/Button.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import PlainLink, { PlainLinkProps } from "./PlainLink";
 
 type ButtonVariant = "normal" | "light" | "transparent" | "transparent-light";
 
-type ButtonProps = PlainLinkProps & { variant?: ButtonVariant };
+type ButtonProps = PlainLinkProps & { variant?: ButtonVariant; type?: "button" | "submit" | "reset" };
```
Comment (pkrucz00) [inline assets/src/features/shared/components/Button.tsx:7]: Can be moved to a different line. Possibly move to `consts.tsx`

File: assets/src/features/shared/components/Input.tsx [hunk 5]
```diff
@@ -62,14 +63,17 @@ const Input: React.FC<InputProps> = ({
...
-        <span
-          className={clsx(
-            "block text-sm",
-            disabled ? "text-brand-grey-80" : error ? "text-brand-red" : "text-brand-dark-blue-500"
-          )}
-        >
-          {additionalText}
-        </span>
+        <div className="text-xm flex items-center gap-1">
+          <Info className={clsx(error ? "stroke-brand-red" : "stroke-brand-dark-blue-500")} />
```
Comment (pkrucz00) [inline assets/src/features/shared/components/Input.tsx:67]: Case for disabled

**Review by pkrucz00 — COMMENTED** [review]
Self-review

File: assets/src/features/leaving-page/components/Questionnaire.tsx [hunk 6]
```diff
@@ -34,44 +54,99 @@ type QuestionnaireProps = {
...
+    register,
+    watch,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<Inputs>();
+  const emailInput = watch("email");
+  const emailFilled = !!emailInput && !errors.email;
+
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
+    console.log(data);
```
Comment (ostatni5) [inline assets/src/features/leaving-page/components/Questionnaire.tsx:68]:
```suggestion
```

File: assets/src/features/leaving-page/components/Questionnaire.tsx [hunk 7]
```diff
@@ -34,44 +54,99 @@ type QuestionnaireProps = {
...
-            <Info />
-            <span className="font-aktivGrotesk text-xs">Information required</span>
-          </div>
+          <Controller
+            name="email"
+            control={control}
+            rules={{
+              pattern: {
+                value:
+                  /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
```
Comment (ostatni5) [inline assets/src/features/leaving-page/components/Questionnaire.tsx:114]: ❤️

Comment (ostatni5) [reply assets/src/features/leaving-page/components/Questionnaire.tsx:114]: Maybe extract to const?

**Pull request merged**